core/dbt/parser/hooks.py

```diff
@@ -72,10 +73,13 @@ def transform(self, node):
 
     # Hooks are only in the dbt_project.yml file for the project
     def get_path(self) -> FilePath:
+        # There ought to be an existing file object for this, but
+        # until that is implemented use a dummy modification time
```

Comment:
If we're using dummy times and the times might not always be accessible, it seems like a good use case for an `Optional` value.

Reply:
This is only for this one use case for creating a temporary FilePath (when we should really have some kind of a file object). Other FilePath objects should always have a modification time, so this should not be optional.  We could read the modification time for this file but that would be a waste since the FilePath is not persistent and is just discarded.

Reply:
To me it just doesn't look like the last updated time is actually guaranteed since the function we use to collect it throws. If it were something we could always have on every os, it wouldn't have a reason to throw.